readme.md
## Hive Mind MitchPlay

--------------------------------------

Take control of the Biters, defend your home planet from the foreign invaders, The Engineers. Build Spawners and influence a pollution based economy, and lead yourself to Victory!

changelog.txt
Version: 0.2.7
Date: 2023-0x-xx

   ~ Minor feature ~
    - added Chinese translations (Thanks Plexpt)

 ~ fixes ~
    - fixed an old bug where unit control spawns biters trough the Popcap. (If Unit Control accepts my update)
---------------------------------------------------------------------------------------------------
Version: 0.2.6
Date: 2023-01-27

 ~ fixes ~
    - fixed compat with ERM
    - fixed compat with space exploration
---------------------------------------------------------------------------------------------------
Version: 0.2.5
Date: 2022-11-12

  ~ fixes ~
    - biters no longer spawn when deconstructed by your own team
    - no crash when equiping armor as a biter
---------------------------------------------------------------------------------------------------
Version: 0.2.4
Date: 2022-10-25

   ~ Minor feature ~
    - added settings for configurable tech costs
    - added settings to change the popcap techs effects per map

  ~ fixes ~
    - modded deployers now can get build once more

 	 ~ Mod Compatibility ~
   	- now compatiable with bobs logistics
    - now compatiable with space exploration
---------------------------------------------------------------------------------------------------
Version: 0.2.3
Date: 2022-10-20
   ~ Mayor feature ~
    - deployers are the same size as the original spawners
    - added research prod for the biters
    - deployers now spawn the biter they where making when they are killed by an enemy

   ~ Minor feature ~
    - fish are now nerfed

 	 ~ Mod Compatibility ~
   	- now compatiable with a future version of RampantFixed
---------------------------------------------------------------------------------------------------
Version: 0.2.2
Date: 2022-10-12

   ~ Minor feature ~
    - A text message tells you how far from spawn you have spawned

 	 ~ Mod Compatibility ~
   		 - now compatiable with enemy race manager
   		 - now compatiable with Arachnids
   		 - now compatiable with humanoids 2
   		 - this may have lead to more compatibilities as well

	~ fixes ~
		- hivemind people can no loner place something that is not yet researched.
    - deployers are now military targets
    - deployers now have the same resistances as the spawners they came from
    - hivemind players spawn the correct distance and not on the edge of the known area
    - hivemind players spawn atleast on a 40 by 40 tiles of land (cliffs are included in that space)
    - other minor issues
---------------------------------------------------------------------------------------------------

Version: 0.2.1
Date: 2022-10-11
Major Features:

	~ Commands ~
		- added a /hivemind command that gives insight into hivemind forces and players

 	 ~ Mod Compatibility ~
   		 - now compatiable with enemy race manager
   		 - now compatiable with natural evolution enemies (fixed)
   		 - this may have lead to more compatibilities as well

	~ fixes ~
		- the single hivemind team map setting now does not crash anymore
    		- popcap is no longer global.
---------------------------------------------------------------------------------------------------


Version: 0.2.0
Date: 2022-10-09
Major Features:

	~ Major Additions ~
		- The Hivemind now has a Working Population Cap
		- The Hivemind now has a Working Pollution Based Tech Tree
		- New Timed Delay for joining the Hivemind
		- Players can no longer Join the Hiveming During the Starting Cutscene (Fixing a lot of bugs caused by it)
		- Lots of Mod Compatitbility Fixes

	~ Building Changes ~
		- blight wall Hp Increased: 45 --> 115
		- blight Spreader Hp Increased: 75 --> 315
		- Armored blight Tumor Hp Increased: 116 --> 215
		- All Worms have been Buffed Slightly to Fit The New Tech Tree

	~ Mod Dependency Changes ~
		- Removed Armored Biters as a Required Dependency
---------------------------------------------------------------------------------------------------
Version: 0.1.7
Date: 2022-05-18
  Major features:
    - Added MitchPlayHiveMindClasses mod as a Dependency; It adds Classes to The Engineer Team
---------------------------------------------------------------------------------------------------
Version: 0.1.6
Date: 2022-05-30
  Major features:
    - Minor Changes to the cost and production of various buildings; Mainliy Pollution Production
	--Pollution Burner Miner Drill: Old Cost 175 --> 200 New Cost
	-- Pollution Mining Drill: Old Cost 350 --> 415 New Cost
	-- Advanced Pollution Mining Drill: Old Cost 1120 --> 1425 New Cost
	-- Pollution oil drill: Old Cost 875 --> 1150 New Cost
	-- Advanced Pollution oil drill: Old Cost 2625 --> 3150 New Cost 
	-- blight Wall: Old Cost 75 -- Old Hp:25 --> New Cost:115 -- New Hp:45
	-- Armoured blight Tumor: Old Hp:125 --> New Hp:165

---------------------------------------------------------------------------------------------------
Version: 0.1.5
Date: 2022-05-18
  Major features:
    - Reworked the Early-Mid Game production of Pollution; Builders should get a little longer to prepare their defenses
    - Added ArmouredBiters as a Dependency (Has Great Compatibility to the mod, Turn off Armored Biter Nests)
    - Fixed Some Capitalization of Some Spelling
---------------------------------------------------------------------------------------------------
Version: 0.1.0
Date: 2022-05-02
  Major features:
    - Initial release


shared.lua
--Shared data interface between data and script, notably prototype names.

local data = {}

data.deployers =
{
  biter_deployer = "biter-deployer",
  spitter_deployer = "spitter-deployer"
}

data.players =
{
  small_biter_player = "small-biter-player",
  medium_biter_player = "medium-biter-player",
  big_biter_player = "big-biter-player",
  behemoth_biter_player = "behemoth-biter-player",
}

data.pollution_proxy = "pollution-proxy"

data.firestarter_gun = "firestarter-gun"
data.firestarter_ammo = "firestarter-ammo"

data.blight = "blight"
data.blight_tumor = "blight-tumor"
data.armored_blight_tumor = "armored-blight-tumor"
data.blight_sticker = "blight-sticker"
data.blight_landmine = "blight-landmine"
data.blight_wall = "blight-wall"
data.blight_chest = "blight-chest"
data.blight_spreader = "blight-spreader"
data.pollution_lab = "pollution-lab"
data.pollution_burner_mining_drill = "pollution-burner-mining-drill"
data.pollution_mining_drill = "pollution-mining-drill"
data.advanced_pollution_mining_drill = "advanced-pollution-mining-drill"
data.pollution_oil_drill = "pollution-oil-drill"
data.advanced_pollution_oil_drill = "advanced-pollution-oil-drill"
data.pollution_tree = "pollution-tree"
data.sticker_proxy = "sticker-proxy"

data.blight_radius = {
  worm_default = 3,
  deployer_default = 9,
  [data.blight_spreader] = 14,
  ["advanced-blight-spreader"] = 30,
  [data.blight_tumor] = 10,
  [data.armored_blight_tumor] = 15,
  [data.blight_wall] = 2,
  ["blight-gate"] = 2
}


data.required_pollution =
{
  [data.deployers.biter_deployer] = 100,
  [data.deployers.spitter_deployer] = 200,
  [data.blight_tumor] = 50,
  [data.armored_blight_tumor] = 415,
  [data.blight_spreader] = 75,
  [data.blight_wall] = 115,
  [data.blight_chest] = 50,
  [data.pollution_lab] = 250,
  [data.pollution_burner_mining_drill] = 200,
  [data.pollution_mining_drill] = 415,
  [data.advanced_pollution_mining_drill] = 1425,
  [data.pollution_oil_drill] = 1150,
  [data.advanced_pollution_oil_drill] = 3150,
  [data.pollution_tree] = 1625,
  ["small-worm-turret"] = 200,
  ["medium-worm-turret"] = 400,
  ["big-worm-turret"] = 800,
  ["behemoth-worm-turret"] = 1600,
}

-- Pollution Burner Miner Drill - 175 cost - 6 pollution 
-- Pollution Mining Drill - 350 cost - 12 pollution 
-- Advanced Pollution Mining Drill - 1120 cost - 36 pollution 
-- Pollution oil drill -875 cost - 45 pollution
-- Advanced Pollution oil drill - 2625 cost - 225 pollution

data.needs_proxy_type =
{
  ["assembling-machine"] = true,
  ["lab"] = true,
  ["mining-drill"] = true
}

data.default_unlocked =
{
  ["small-biter"] = true,
  --["small-spitter"] = true,
  ["small-worm-turret"] = true
}

data.needs_tech =
{
  [data.blight_wall] = {},
  [data.armored_blight_tumor] = {"hivemind-unlock-"..data.blight_wall},
  [data.blight_spreader] = {"hivemind-unlock-"..data.blight_wall},
  [data.blight_chest] = {},
  [data.pollution_mining_drill] = {},
  [data.advanced_pollution_mining_drill] = {"hivemind-unlock-"..data.pollution_mining_drill, "hivemind-unlock-"..data.blight_chest},
  [data.pollution_oil_drill] = {"hivemind-unlock-"..data.pollution_mining_drill, "hivemind-unlock-"..data.blight_chest},
  [data.advanced_pollution_oil_drill] = {"hivemind-unlock-"..data.pollution_oil_drill, "hivemind-unlock-"..data.advanced_pollution_mining_drill},
  [data.pollution_tree] = {"hivemind-unlock-"..data.blight_wall},
}

data.needs_oponent_tech =
{
  --hivemind = {
  --  ["hivemind-unlock-"..data.blight_wall] = {"stone-wall"},
  --  ["hivemind-unlock-"..data.pollution_oil_drill] = {"oil-processing"},
  --  ["hivemind-unlock-"..data.advanced_pollution_oil_drill] = {"advanced-oil-processing"},
  --  ["hivemind-unlock-"..data.advanced_pollution_mining_drill] = {"mining-productivity-1"}
  --}
}

data.needs_blight =
{
  ["small-worm-turret"] =true,
  ["medium-worm-turret"] = true,
  ["big-worm-turret"] = true,
  ["behemoth-worm-turret"] = true,
  [data.blight_tumor] = true,
  [data.blight_wall] = true,
  [data.armored_blight_tumor] = true,
  [data.pollution_mining_drill] = true,
  [data.advanced_pollution_mining_drill] = true,
  [data.pollution_burner_mining_drill] = true,
  [data.pollution_oil_drill] = true,
  [data.advanced_pollution_oil_drill] = true,
  [data.pollution_lab] = true
}

data.pollution_cost_multiplier = 1
data.deployer_speed_modifier = 0.25
data.spawning_attempts_per_radius = 30 --it schrinks the spawning radius 5 times (83%, 67%, 50%, 33% and 17%) and attempts to spawn the biter player in that radus

data.evolution_factor_to_pollution_cost = --pollution_cost = base + round(multiplier *((round(evolution_factor,1)*10)^(power_effect * evolution_factor)),0) * 25
{
  base = -300,       --this helps the starter values to be balanced
  multiplier = 19.8, --this helps give teh numbers a good spread
  power_effect = 1   --higher numbers give a bigger gap between the earliest and the highest
}

data.summon_starter_data = {
  [data.players.small_biter_player] = {
    ["units"] = {
      ["small-biter"] = 20,
      ["small-spitter"] = 10,
      --["small-builder"] = 10
    },
    ["default_biters_and_worm_unlocked_factor"] = 0
  },
  [data.players.medium_biter_player] = {
    ["units"] = {
      ["small-biter"] = 12,
      ["small-spitter"] = 6,
      ["medium-biter"] = 4,
      ["medium-spitter"] = 3
    },
    ["default_biters_and_worm_unlocked_factor"] = 0.35
  },
  [data.players.big_biter_player] = {
    ["units"] = {
      ["medium-biter"] = 17,
      ["medium-spitter"] = 10,
      ["big-biter"] = 8,
      ["big-spitter"] = 3
    },
    ["default_biters_and_worm_unlocked_factor"] = 0.7
  },
  [data.players.behemoth_biter_player] = {
    ["units"] = {
      ["big-biter"] = 14,
      ["big-spitter"] = 5,
      ["behemoth-biter"] = 2,
      ["behemoth-spitter"] = 2
    },
    ["default_biters_and_worm_unlocked_factor"] = 1
  }
}

return data
